### DISCUSSION TITLE
Is it possible to use Prisma Client for a particular entity?

### DISCUSSION THREAD

**Discussion by simonuvarov** [prisma/prisma#4321] [Q&A] [answered]
Hi everyone, 

Let's say I have images and tags and I also have a generic function like `getPaginated{Images|Tags}` that calls `prisma.{images|tags}.findMany` inside. Currently, I have to write the same function two times even though the only difference is which entity I use. So, my question is: is it possible to pass something like `prisma.image` or `prisma.tag` to a function instead of `prisma` of type `PrismaClient`? Or maybe there's a better way to reuse such logic.

It might be possible to use `ModelName` enum and then lowercase it and use Prisma Client as `prisma[ModelName].findMany()`, but it does not look like a perfect solution.

Thanks,
Simon

**Comment by ryands17** [comment, marked as answer]
@simonuvarov 👋 

The above approach is not recommended as you wont be able to type the output correctly and if you're using TypeScript, then it will complain about the return type. It's best the keep them separate even though the paramters are the same as if you want to extend something later, this might become a problem and also the types would get messed up. Hope this helps :)

**Reply by simonuvarov** [reply]
Yes, I must admit that I don't like this approach and currently I'm just copying this function and change the entity name in it, so I guess you're right.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
